src/java.base/share/classes/java/lang/invoke/InvokerBytecodeGenerator.java

```diff
@@ -131,9 +135,17 @@ private InvokerBytecodeGenerator(LambdaForm lambdaForm, int localsMapSize,
             name = makeDumpableClassName(name);
         }
         this.name = name;
-        this.className = CLASS_PREFIX.concat(name);
-        validateInternalClassName(name);
-        this.classEntry = pool.classEntry(ReferenceClassDescImpl.ofValidated(concat("L", className, ";")));
+        var classDesc = switch (name) {
+            case "MH"  -> CD_MH;
+            case "DMH" -> CD_DMH;
+            case "BMH" -> CD_BMH;
```

Comment:
I don't see where DMH and BMH are passed; only see NFI and LFI.

Reply:
This DMH is derived from invokerName, which comes from java.lang.invoke.LambdaForm.Kind#defaultLambdaName

Reply:
`"NFI"` and `"LFI"` should probably also be added to this `switch`.

Reply:
No: I believe these names are used for invokers spun exclusively for the LF interpreter, which is only used for debugging (and as a development aide) these days. Micro-optimizing those code paths only add complexity for no gain in practice.